### PR TITLE
Crash with gestures

### PR DESCRIPTION
**Pull request by zwim**
This is the successor of "Fix crash with "open last document" gesture and no lastfile"  #7691

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/7716)

### CODE REVIEW

**Comment by zwim** [comment]
I had some git troubles with #7691 . A `git push origin ...` gave me a new PR (after a rebase). Should I have done 'git push -u origin ...`?

**Comment by poire-z** [comment]
> A git push origin ... gave me a new PR (after a rebase). Should I have done 'git push -f origin ...`?

Dunno, a push after a rebase without -f should have failed.
If you got a new PR, may be the branch had another name ?
Previous PR was `zwim:fixCrashGesture` - this PR is `zwim:crashWithGestures`, so looks like that.

File: frontend/dispatcher.lua [hunk 1]
```diff
@@ -617,36 +617,38 @@ arguments are:
 function Dispatcher:execute(ui, settings, gesture)
     for k, v in pairs(settings) do
         if settingsList[k] ~= nil and (settingsList[k].conditions == nil or settingsList[k].conditions == true) then
-            if settingsList[k].category == "none" then
-                if settingsList[k].arg ~= nil then
-                    ui:handleEvent(Event:new(settingsList[k].event, settingsList[k].arg))
-                else
-                    ui:handleEvent(Event:new(settingsList[k].event))
+            if (not settingsList[k].paging and not settingsList[k].rolling) or self.document then
```
Comment (poire-z) [inline frontend/dispatcher.lua:620]: May be add the comment I suggested in the other PR, so we know what's that about :)
`-- Be sure we don't send a document setting event if there's not yet or not longer a document
`

**Comment by zwim** [comment]
> > A git push origin ... gave me a new PR (after a rebase). Should I have done 'git push -f origin ...`?
> 
> Dunno, a push after a rebase without -f should have failed.
> If you got a new PR, may be the branch had another name ?
> Previous PR was `zwim:fixCrashGesture` - this PR is `zwim:crashWithGestures`, so looks like that.

Ohh, indeed. I pushed the thing :/

But the content of this PR is ok.

File: frontend/dispatcher.lua (hunk 1 above)
Comment (poire-z) [reply frontend/dispatcher.lua:621]: Oups, my bad english, should be "no longer", not "not longer" as @NiLuJe corrected in the other PR :)

Comment (Frenzie) [reply frontend/dispatcher.lua:621]: That's a comment to a previous commit? It reads a bit confusing.

**Review by poire-z — APPROVED** [review]

File: frontend/dispatcher.lua (hunk 1 above)
Comment (Frenzie) [reply frontend/dispatcher.lua:621]: I would probably flip the condition for clarity, or is that just me?

Comment (poire-z) [reply frontend/dispatcher.lua:621]: > That's a comment to a previous commit? It reads a bit confusing.

Initially at https://github.com/koreader/koreader/pull/7691#discussion_r635419091

File: frontend/dispatcher.lua [hunk 2]
```diff
@@ -618,7 +618,7 @@ function Dispatcher:execute(ui, settings, gesture)
     for k, v in pairs(settings) do
         if settingsList[k] ~= nil and (settingsList[k].conditions == nil or settingsList[k].conditions == true) then
             -- Be sure we don't send a document setting event if there's not yet or no longer a document
-            if (not settingsList[k].paging and not settingsList[k].rolling) or self.document then
+            if self.document or (not settingsList[k].paging and not settingsList[k].rolling) then
```
Comment (poire-z) [inline frontend/dispatcher.lua:621]: You're sure it works with `self.document` ?  The other reference in this file is `ui.document`.

Comment (zwim) [reply frontend/dispatcher.lua:621]: Yes I have tripple checked: With `self.document` and `ui.document` the patch works. With none of them I get the crash.

So for consistancy I think we should use `ui.document` here too.

Comment (poire-z) [reply frontend/dispatcher.lua:621]: Well, I don't see anywhere some code that would set or update `Dispatcher.document` :/
So, best to go with `ui.document` (as `ui` is provided to the function call, so we're sure it's up to date.

**Review by Frenzie — APPROVED** [review]

**Pull request merged**